test/withDataProviders.test.js

```diff
@@ -0,0 +1,145 @@
+/* global test, expect, beforeEach, describe */
+/* eslint-disable no-console */
+import React from 'react'
+import ReactDOM from 'react-dom'
+import {withDataProviders} from '../src/withDataProviders'
+import {compose} from 'redux'
+import {connect} from 'react-redux'
+import {newTestApp, getData, getDataWithCount, GET_DATA_DELAY} from './common'
+import {Promise} from 'bluebird'
+
+beforeEach(() => {
+  getDataWithCount.counter = 0
+})
+
+test('withDataProviders (needed=true) renders child component when the data is fetched', async () => {
+  const {root} = renderMessageContainerApp({needed: true})
+
+  let renderedMessage = root.querySelector('div.message p')
+  expect(renderedMessage).toBeNull()
+
+  await Promise.delay(GET_DATA_DELAY)
+  renderedMessage = root.querySelector('div.message p')
+  expect(renderedMessage).not.toBeNull()
+  expect(renderedMessage.textContent).toBe('Hello')
+})
+
+test('withDataProviders (needed=false) renders initial state of child component', async () => {
+  const {root} = renderMessageContainerApp({needed: false, initialData: {data: 'init'}})
+
+  let renderedMessage = root.querySelector('div.message p')
+  expect(renderedMessage).not.toBeNull()
+  expect(renderedMessage.textContent).toBe('init')
+
+  await Promise.delay(GET_DATA_DELAY)
+  renderedMessage = root.querySelector('div.message p')
+  expect(renderedMessage).not.toBeNull()
+  expect(renderedMessage.textContent).toBe('Hello')
+})
+
+describe('nested withDataProviders with eager prefetching', () => {
+  const getData = [getDataWithCount, {data: ''}, GET_DATA_DELAY]
```

Comment:
Tests should be isolated from one another. Running a test as `test.only` should be no different from running it as part of a long test suite. If you need some setup before each of those tests, I'd suggest creating a function, which is ran in each test, does all the work and returns the necessary variables.

Reply:
Well, the way I structured this test was that I included the setup code in the `describe` block and each `test` block rely on this setup - but they can still be skipped or ran independently -> both `.skip` and `.only` work here (on both `describe` and `test` blocks)

Reply:
I've flatten this test under a single `test()`, because I've removed the other test here, and therefore the describe-grouping was not needed here anymore.